metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchRequestHandler.java

```diff
@@ -100,8 +101,13 @@ public static SearchRequestHandler getBuilder(@Nonnull EntitySpec entitySpec) {
 
   public static BoolQueryBuilder getFilterQuery(@Nullable Filter filter) {
     BoolQueryBuilder filterQuery = ESUtils.buildFilterQuery(filter);
-    // Filter out entities that are marked "removed"
-    filterQuery.mustNot(QueryBuilders.matchQuery("removed", true));
+
+    // Filter out entities that are marked "removed" if and only if filter does not contain a criterion referencing removed
+    if (filter == null || (filter.hasCriteria() && filter.getCriteria().stream()
+            .anyMatch(criterion -> criterion.getField().equals(REMOVED)))) {
```

Comment:
Shouldn't this be the negation of
```java
filter.getCriteria().stream().anyMatch(criterion -> criterion.getField().equals(REMOVED))
```
?

Reply:
Please add tests & update https://github.com/linkedin/datahub/blob/master/metadata-io/src/test/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchRequestHandlerTest.java.

Reply:
`Filter.getCriteria()` is deprecated: https://github.com/linkedin/datahub/blob/5aaf187371902e03318220c8c1b9b68ea3e999c3/metadata-models/src/main/pegasus/com/linkedin/metadata/query/filter/Filter.pdl#L14

Using only criterions from `or` property.

Will add tests.

Reply:
Correct!